connectors/sources/azure_blob_storage.py

```diff
@@ -190,19 +191,38 @@ async def get_content(self, blob, timestamp=None, doit=None):
             document,
             filename,
             file_extension,
-            partial(self.blob_download_func, filename, blob["container"]),
+            partial(self.blob_download_func, filename, blob["container"], file_size),
         )
 
-    async def blob_download_func(self, blob_name, container_name):
-        async with BlobClient.from_connection_string(
-            conn_str=self.connection_string,
-            container_name=container_name,
-            blob_name=blob_name,
-            retry_total=self.retry_count,
-        ) as blob_client:
-            data = await blob_client.download_blob()
-            async for content in data.chunks():
-                yield content
+    def _get_container_client(self, container_name):
+        if self.container_clients.get(container_name) is None:
+            self.container_clients[
+                container_name
+            ] = ContainerClient.from_connection_string(
+                conn_str=self.connection_string,
+                container_name=container_name,
+                retry_total=self.retry_count,
+            )
+            return self.container_clients[container_name]
+        else:
+            return self.container_clients[container_name]
+
+    async def blob_download_func(self, blob_name, container_name, file_size):
+        container_client = self._get_container_client(container_name=container_name)
+        offset = 0
+        length = 100000
```

Comment:
What is this number? Why 100000? Can we put it into a meaningful constant or even a setting?

Reply:
It's used to fetch the file content in the chunks of 100 Kb. I've put it in a constant.